### PR TITLE
Fixed #137

### LINKED ISSUE #137: [OperatorPrecedence] Use grouping parenthesis to make the operator precedence explicit
In Epoxy Processor 
`builder.beginControlFlow("if ($L != null && that.$L == null"
                + " || $L == null && that.$L != null)",`
No Operator Precedence exists
[Reference](http://errorprone.info/bugpattern/OperatorPrecedence)

### PR DESCRIPTION
**Pull request by engincancan**
Used grouping parenthesis to make the operator precedence explicit

### CODE REVIEW

File: epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java [hunk 1]
```diff
@@ -662,8 +662,8 @@ private MethodSpec generateEquals(ClassToGenerateInfo helperClass) {
               .endControlFlow();
         }
       } else {
-        builder.beginControlFlow("if ($L != null && that.$L == null"
-                + " || $L == null && that.$L != null)",
+        builder.beginControlFlow("if (($L != null && that.$L == null)"
```
Comment (elihart) [inline epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java]: I believe we could also simplify this to `if(($L == null) != (that.$L == null))` if you're so inclined

Comment (engincancan) [inline epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java]: It is a simply XOR operation. I didn't quite catch that. Thanks for pointing out :) I will make the change and update it.

**Comment by elihart** [comment]
Thanks!

I'm in the middle of a very large refactor and update of the processor (see https://github.com/airbnb/epoxy/pull/139) so I'm not actually going to merge your PR because that will cause me a bunch of merge headaches. I will incorporate your change into the processor update though (doing it now! :) )

Thanks again!

**Comment by elihart** [comment]
Your change is incorporated into https://github.com/airbnb/epoxy/pull/140 :)

https://github.com/airbnb/epoxy/pull/140/files#diff-e7b2767f9be47977fb1b4c944e9919d2R599

**Pull request closed without merging**